src/Nest/IndexModules/IndexSettings/Settings/AutoExpandReplicas.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	[JsonConverter(typeof(AutoExpandReplicasJsonConverter))]
+	public class AutoExpandReplicas
+	{
+		private const string AllMaxReplicas = "all";
+		private bool _enabled;
+		private Union<int?, string> _maxReplicas;
+		private int? _minReplicas;
+
+		public bool Enabled => _enabled;
+
+		public int? MinReplicas
+		{
+			get { return _minReplicas; }
+			set
+			{
+				if (value == null && _maxReplicas == null) _enabled = false;
+				else _enabled = true;
+				_minReplicas = value;
+			}
+		}
+
+		public Union<int?, string> MaxReplicas
+		{
+			get { return _maxReplicas; }
+			set
+			{
+				if (value == null && _minReplicas == null) _enabled = false;
+				else _enabled = true;
+				_maxReplicas = value;
+			}
+		}
+
+		public static AutoExpandReplicas Disabled() => new AutoExpandReplicas();
```

Comment:
`AutoExpandReplicas` looks immutable to me so this could be a static property `{ get; } = new AutoExpandReplicas()` so that single instance could be reused.